docs/HTTP_SPEC.md

```diff
@@ -14,107 +14,287 @@
+Same as [Trustless Gateway](https://specs.ipfs.tech/http-gateways/trustless-gateway/#accept-request-header), but _SHALL NOT_ accept any other media types specified other than `application/vnd.ipld.car`.
 
-##### Path Parameters
+Used to specify the response content type. _OPTIONAL_ only if a `format` query parameter is provided, otherwise this header is _REQUIRED_.
 
-- `cid` - _Required_. A valid string representation of the root CID of the DAG being requested.
+If provided, the value MUST explicitly or implicitly include the following media type:
+- `application/vnd.ipld.car`
 
-- `path` - _Optional_. A valid IPLD path to traverse within the DAG to the final content.
+#### `version` (CAR content type parameter)
 
-    The path must begin with a `/`, and must describe a valid path within the DAG. The path will be resolved as a [UnixFS](https://github.com/ipfs/specs/blob/main/UNIXFS.md) path where the encountered path segments are within valid UnixFS blocks and can be read as named links. Where the blocks do not describe valid UnixFS data, the path segment(s) will be interpreted as describing plain IPLD nodes to traverse.
+See [application/vnd.ipld.car](https://www.iana.org/assignments/media-types/application/vnd.ipld.car).
+
+_OPTIONAL_. `version=1`. Defaults to `1`.
+
+Used to specify the version of the CAR media type to respond with. Values other than `1` will respond with a 400 status code.
+
+- `1`: Version 1 of the CAR media type.
+
+Example:
+- `application/vnd.ipld.car;version=1;` requests version 1 of the CAR media type
+
+#### `dups` (CAR content type parameter)
+
+Specified in [IPIP-412](https://github.com/ipfs/specs/blob/746cb5206e2346f8e4a37219898a735a498c3186/src/ipips/ipip-0412.md#order-car-content-type-parameter).
+
+_OPTIONAL_. `dups=(y|n)`. Defaults to `y`. 
+
+Used to specify whether or not the response will include duplicate blocks in the CAR response. Unspecified values will respond with a 400 status code.
+
+- `y`: Include duplicate blocks in the response
+- `n`: Does not include duplicate blocks in the response
+
+Examples:
+- `application/vnd.ipld.car;dups=y;` includes duplicate blocks
+- `application/vnd.ipld.car;dups=n;` does not include duplicate blocks
+
+#### `order` (CAR content type parameter)
+
+Specified in [IPIP-412](https://github.com/ipfs/specs/blob/746cb5206e2346f8e4a37219898a735a498c3186/src/ipips/ipip-0412.md#order-car-content-type-parameter), but Lassie will only ever produce a `dfs` ordered CAR response.
+
+_OPTIONAL_. `order=(dfs|unk)`. Defaults to `unk`.
+
+Used to specify preference for a specific block order in the CAR response. Unspecified values will respond with a 400 status code.
+
+- `dfs`: [Depth-First Search](https://en.wikipedia.org/wiki/Depth-first_search) order, enables streaming responses with minimal memory usage. 
+- `unk`: Unknown order. Although this is the default, Lassie will still produce a depth first search regardless.
+
```

Comment:
```suggestion
- `dfs`: [Depth-First Search](https://en.wikipedia.org/wiki/Depth-first_search) order.
- `unk`: Unknown order. Although this is the default, Lassie will still produce a depth first search regardless.

Depth-first traversal enables efficient incremental verifiability of content, as the consume of the DAG can execute the same traversal on the data as it is received, verifying the content is exactly what is expected and rejecting the content as soon as a discrepancy occurs rather than waiting to consume the entire DAG before being able to reconstruct the ordered content.

When combined with `dups=y`, a depth-first traversal enables streaming responses, obviating the need to cache blocks in case they re-appear later in a traversal.

```

Hopefully that's not too wordful but I wanted to add the benefits of dfs and clear up that dfs-alone doesn't give you efficient streaming.